mock-tikv/src/pd.rs

```diff
@@ -0,0 +1,313 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::{spawn_unary_success, MOCK_TIKV_PORT};
+use futures::{FutureExt, StreamExt, TryFutureExt};
+use grpcio::{Environment, Server, ServerBuilder, WriteFlags};
+use kvproto::pdpb::*;
+use std::sync::Arc;
+
+pub const MOCK_PD_PORT: u16 = 50021;
+/// This is mock pd server, used with mock tikv server.
+#[derive(Debug, Clone)]
+pub struct MockPd {
+    ts: i64,
+}
+
+impl MockPd {
+    fn new() -> MockPd {
+        MockPd { ts: 0 }
+    }
+
+    fn region() -> kvproto::metapb::Region {
+        let mut meta_region = kvproto::metapb::Region::default();
+        meta_region.set_end_key(vec![0xff; 20]);
+        meta_region.set_start_key(vec![0x00]);
+        meta_region.set_id(0);
+        meta_region.set_peers(vec![Self::leader()]);
+        meta_region
+    }
+
+    fn leader() -> kvproto::metapb::Peer {
+        kvproto::metapb::Peer::default()
+    }
+
+    fn store() -> kvproto::metapb::Store {
+        let mut store = kvproto::metapb::Store::default();
+        store.set_address(format!("localhost:{}", MOCK_TIKV_PORT));
+        // TODO: start_timestamp?
+        store
+    }
+}
+
+pub fn start_mock_pd_server() -> Server {
+    let env = Arc::new(Environment::new(1));
+    let mut server = ServerBuilder::new(env)
+        .register_service(create_pd(MockPd::new()))
+        .bind("localhost", MOCK_PD_PORT)
+        .build()
+        .unwrap();
+    server.start();
+    server
+}
+
+impl Pd for MockPd {
+    fn get_members(
+        &mut self,
+        ctx: ::grpcio::RpcContext,
+        req: GetMembersRequest,
+        sink: ::grpcio::UnarySink<GetMembersResponse>,
+    ) {
+        let mut resp = GetMembersResponse::default();
+        resp.set_header(ResponseHeader::default());
+        let mut member = Member::default();
```

Comment:
Are the fields public? I think it is easier to use a struct literal rather than default and setters if they are. You can use the following pattern if you only want to initialize some fields:
```
let f = Foo {
    a: whatever,
    b: something,
    ..Foo::default()
};
```